src/proc/threshold.cpp

```diff
@@ -0,0 +1,64 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include "../include/librealsense2/hpp/rs_sensor.hpp"
+#include "../include/librealsense2/hpp/rs_processing.hpp"
+
+#include "proc/synthetic-stream.h"
+#include "context.h"
+#include "environment.h"
+#include "option.h"
+#include "threshold.h"
+#include "image.h"
+
+namespace librealsense
+{
+    threshold::threshold() : _min(0.15f), _max(4.f)
+    {
+        auto min_opt = std::make_shared<ptr_option<float>>(0.f, 16.f, 0.1f, 0.15f, &_min, "Min range in meters");
+        register_option(RS2_OPTION_MIN_DISTANCE, min_opt);
```

Comment:
Consider change to RS2_OPTION_MIN_Z in case we will add x and y in the future

Reply:
RS2_OPTION_MIN_Z that would be a new option, prefer to reuse the options we already have.
For x,y "thresholding" I'd avoid options, and use `roi_sensor` extension (maybe modify it a bit). 
Otherwise you have options with floating range.